### ISSUE TITLE
Integrate typed-css-modules 

### ISSUE THREAD

**Issue by mohsen1** [fathyb/parcel-plugin-typescript#47]
Would you accept a pull request to add support for [typed-css-modules](https://github.com/Quramy/typed-css-modules) or you think that should be a different plugin on its own?

Related to https://github.com/Quramy/typed-css-modules/issues/51

**Comment by fathyb** [comment]
👍 it would be more than welcome, let me know if you need any help

**Comment by Place1** [comment]
I've implemented this over at https://github.com/place1/parcel-plugin-typed-css-modules

**Comment by mohsen1** [comment]
That's great! I believe Parcel has of not actually writing to disk for this. @fathyb Can you confirm if that's true?

**Comment by Place1** [comment]
@mohsen1 I can’t understand what you’re asking 😬

**Comment by mohsen1** [comment]
I mean those `.d.ts` files. Can we avoid writing those on disk

**Comment by Place1** [comment]
right. i chose to write them to the file system because that's what [webpack typed css modules loader](https://github.com/olegstepura/typed-css-modules-loader) does.

importantly, it means the TS compiler and IDEs can find the declaration files. This is very useful because it means the language server can offer automatic import hints when typing css class names in ts files.

with that said, i haven't really thought about an alternative, do you have something in mind?

**Comment by mohsen1** [comment]
I think I remember now. Parcel is not using a memory fs yet. Once Parcel supports memory fs we can "write" to that fs without actually messing up the actual fs. TS language server does support custom fs. that's how it works in browser. 

I guess if you avoid writing to actual disk then you need to run a custom language server for your IDE as well. Maybe it doesn't worth it.
